code/game/objects/effects/spawners/random/structure.dm

```diff
@@ -0,0 +1,201 @@
+		/obj/structure/girder/displaced = 1,
+	)
+
+/obj/effect/spawner/random/structure/grille
+	name = "grille spawner"
+	icon_state = "grille"
+	loot = list(
+		/obj/structure/grille = 4,
+		/obj/structure/grille/broken = 1,
+	)
+
+/obj/effect/spawner/random/structure/lattice
+	name = "lattice spawner"
+	icon_state = "lattice"
+	loot = list(
+		/obj/structure/lattice = 4,
+		/obj/structure/lattice/catwalk = 1,
+	)
+
+/obj/effect/spawner/random/structure/spare_parts
+	name = "spare parts spawner"
+	icon_state = "table_parts"
+	loot = list(
+		/obj/structure/table_frame,
+		/obj/structure/table_frame/wood,
+		/obj/item/rack_parts,
+	)
+
+/obj/effect/spawner/random/structure/table_or_rack
+	name = "table or rack spawner"
+	icon_state = "rack_parts"
+	loot = list(
+		/obj/effect/spawner/random/structure/table,
+	)
+
+/obj/effect/spawner/random/structure/table
+	name = "table spawner"
+	icon_state = "table"
+	loot = list(
+		/obj/structure/table = 40,
+		/obj/structure/table/wood = 30,
+		/obj/structure/table/glass = 20,
+		/obj/structure/table/reinforced = 5,
+		/obj/structure/table/wood/poker = 5,
+	)
+
+/obj/effect/spawner/random/structure/table_fancy
+	name = "table spawner"
+	icon_state = "table_fancy"
+	loot_type_path = /obj/structure/table/wood/fancy
+	loot = list()
+
+/obj/effect/spawner/random/structure/tank_holder
+	name = "tank holder spawner"
+	icon_state = "tank_holder"
+	loot = list(
+		/obj/structure/tank_holder/oxygen = 40,
+		/obj/structure/tank_holder/extinguisher = 40,
+		/obj/structure/tank_holder = 20,
+		/obj/structure/tank_holder/extinguisher/advanced = 1,
+	)
+
+/obj/effect/spawner/random/structure/crate_empty
+	name = "empty crate spawner"
+	icon_state = "crate"
+	loot = list(
+		/obj/structure/closet/crate = 20,
+		/obj/structure/closet/crate/wooden = 1,
+		/obj/structure/closet/crate/internals = 1,
+		/obj/structure/closet/crate/medical = 1,
+		/obj/structure/closet/crate/freezer = 1,
+		/obj/structure/closet/crate/radiation = 1,
+		/obj/structure/closet/crate/hydroponics = 1,
+		/obj/structure/closet/crate/engineering = 1,
+		/obj/structure/closet/crate/engineering/electrical = 1,
+		/obj/structure/closet/crate/science = 1,
+	)
+
+/obj/effect/spawner/random/structure/crate_loot
+	name = "lootcrate spawner"
+	icon_state = "crate"
+	loot = list(
+		/obj/structure/closet/crate/maint = 15,
+		/obj/effect/spawner/random/structure/crate_empty = 4,
+		/obj/structure/closet/crate/secure/loot = 1,
+	)
+
+/obj/effect/spawner/random/structure/closet_private
+	name = "private closet spawner"
+	icon_state = "cabinet"
+	loot = list(
+		/obj/structure/closet/secure_closet/personal,
+		/obj/structure/closet/secure_closet/personal/cabinet,
+	)
+
+/obj/effect/spawner/random/structure/closet_empty
+	name = "empty closet spawner"
+	icon_state = "locker"
+	loot = list(
+		/obj/structure/closet = 850,
+		/obj/structure/closet/cabinet = 150,
+		/obj/structure/closet/acloset = 1,
+	)
+
+/obj/effect/spawner/random/structure/closet_hallway
```

Comment:
I have a bit of a concern with this one: ending up with Dilbert RNG spam of one or the other type of closet where both are really necessary.
![dilbert](https://user-images.githubusercontent.com/58124831/134414298-bb5e3344-f77b-401e-9cd5-c887fb925568.gif)

Reply:
Removed with https://github.com/tgstation/tgstation/pull/60522/commits/f63c675013bfcf843b0d5257f5d7c6c520ada790